File: shoprbackend/src/main/java/com/realdolmen/shoprbackend/domain/Game.java
package com.realdolmen.shoprbackend.domain;

import javax.persistence.*;
import javax.validation.constraints.Size;

@Entity
@Table(name = "Games")
public class Game extends Item {


    @Size(max = 100)
    private String publisher;

    @Column(name="min_age")
    private short minAge;

//    @Transient
//    @Enumerated(EnumType.STRING)
//    private GameGenre genre;


    public String getPublisher() {
        return publisher;
    }

    public void setPublisher(String publisher) {
        this.publisher = publisher;
    }

    public short getMinAge() {
        return minAge;
    }

    public void setMinAge(short minAge) {
        this.minAge = minAge;
    }

//    public GameGenre getGenre() {
//        return genre;
//    }
//
//    public void setGenre(GameGenre genre) {
//        this.genre = genre;
//    }
//
//
//    @Override
//    public String toString() {
//        return "Game{" +
//                "publisher='" + publisher + '\'' +
//                ", minAge=" + minAge +
//                ", genre=" + genre +
//                '}';
//    }
}

File: shoprFrontend/src/app/models/game.ts
import {Item} from "./item";

export class Game extends Item{

  publisher:string
  minAge:number
  gameGenre:string


  constructor( id?:number, title?: string, type?: string, price?: number, publisher?: string, minAge?: number, gameGenre?:string) {
    super(id, title, type, price);
    this.publisher = publisher;
    this.minAge = minAge;
    this.gameGenre=gameGenre;
  }
}
